Source/OCMock/OCMInvocationStub.m

```diff
@@ -58,27 +55,31 @@ - (void)handleInvocation:(NSInvocation *)anInvocation
     BOOL isInCreateFamily = isInInitFamily ? NO : [anInvocation methodIsInCreateFamily];
     if(isInInitFamily || isInCreateFamily)
     {
-        id returnVal = UNSET_RETURN_VALUE_MARKER;
-        [anInvocation setReturnValue:&returnVal];
+        id returnValBeforeInvocation = [[NSObject alloc] init];
```

Comment:
Instead of using a real object, have you tried to use a fake pointer value that is divisible by 8? That would be an easier change and it might just work. Would try it but I don't have access to non-Intel hardware.

Reply:
I tried now a bunch of different addresses: 0x1000000,  0x100000, 0x10000000, 0x10 etc. Can't find a value that would not crash. Isn't it safer to use real object? It ensures it can not crash on any arch, existing or future.

Reply:
@erikdoe I was advised to use the following trick to ensure that the address is not only valid but also mapped:

```
static const void *kValidAddress = &kValidAddress;
``` 

This still crashes, though in a different place. It is worth mentioning that the crashes are in `objc_retain`. 

What are your concerns with using a real object to serve as a marker?

Reply:
<img width="1290" alt="Screen Shot 2022-01-29 at 23 20 07" src="https://user-images.githubusercontent.com/652603/151679873-b6bb864c-628b-4f57-a522-952a70a964b6.png">
<img width="1322" alt="Screen Shot 2022-01-29 at 22 41 10" src="https://user-images.githubusercontent.com/652603/151679876-27322b8b-40ce-470b-8070-1de060c4769d.png">